### DISCUSSION TITLE
Can we run lighthouse on desktop application which are built using Chromium Embedded Framework (CEF) via node?

### DISCUSSION THREAD

**Discussion by ajitesh11** [GoogleChrome/lighthouse#15462] [Q&A]
Can we run lighthouse on desktop application which are built using Chromium Embedded Framework (CEF) Programmatically ? Via node for eg.
If yes How?
If yes, can we do that via lighthouse-ci cli (https://www.npmjs.com/package/@lhci/cli) too?

I did a lot of research and I stuck at a point where i realize EITHER i need to pass a url to lighthouse OR pass --collect.staticDistDir to tell Lighthouse CI where your HTML files are located.

In my case none of the above is possible because the url of application is embedded in a desktop application.

Yes i can run lighthouse on that url in the browser but my requirement is not to run in the browser but on the desktop application which we have built for the application.

Any lead or input in this regard will be greatly appreciated.

**Comment by adamraine** [comment]
> Can we run lighthouse on desktop application which are built using Chromium Embedded Framework (CEF) Programmatically ?

I'm not sure if this is possible, and it's definitely not a workflow we officially support. Lighthouse is designed to work with the actual browser not an embedded framework.
